### ISSUE TITLE
shell/exec example

### ISSUE THREAD

**Issue by tomqwpl** [gliderlabs/ssh#199]
At first sight, the obvious example that is missing is one that does what you expect SSH todo, to run the command supplied by exec, or to create a shell. 
In fact I was sort of expecting that functionality to be built in, rather than being an example.
I can see there is an example that runs a docker command, and an example that runs a fixed command "top" in a pty, but nothing that would do what I would naively expect an ssh server to do by default.

Is the reason one isn't provided that it's more complicated than that, or that one just hasn't been written?

Thanks.

**Comment by perbu** [comment]
I've written quite a few ssh servers and I've never needed to do shell/exec. Doing shell/exec is a security risk. If I'd need to do so I'd probably use OpenSSH instead.

Ssh is a great way to make a terminal app available via the network. It's also a reasonably robust way of forwarding TCP based services. So I guess that is why this exists.

**Comment by gustavosbarreto** [comment]
> @tomqwpl Is the reason one isn't provided that it's more complicated than that, or that one just hasn't been written?

No, there is no problem with executing commands if it makes sense for your application that serves SSH. You should be responsible for handling authentication yourself to ensure that there are no security implications. There is no example yet because no one has created one.

> @perbu Doing shell/exec is a security risk. If I'd need to do so I'd probably use OpenSSH instead.

Doing a shell/exec is not a security risk if you provide a secure authentication method in your application. The intention of this package is not to replace OpenSSH, but if you want to provide the same functionality as OpenSSH, there is no problem with that.

**Comment by gustavosbarreto** [comment]
If any of you are looking for how to do authentication using Linux shadow, you can see how we did it in:
1) https://github.com/shellhub-io/shellhub/tree/master/agent/pkg/osauth
2) https://github.com/shellhub-io/shellhub/tree/master/agent/pkg/yescrypt

**Comment by abakum** [comment]
https://github.com/gliderlabs/ssh/pull/214

**Comment by CHN-STUDENT** [comment]
@abakum hi abakum, i use yours fork project ssh-combo to test.  
And i have question, why not you save public key by using `gossh.NewPublicKey(&privateKey.PublicKey)` to save as openssh key-pair?

**Comment by CHN-STUDENT** [comment]
here my code: https://github.com/CHN-STUDENT/ssh/blob/master/_examples/ssh-combo/main.go i hope get more advice!

**Comment by abakum** [comment]
@CHN-STUDENT Hi, look complex example of ssh client and server with certificate autorisation https://github.com/abakum/ngrokSSH
